### PR TITLE
allow users to override the URL

### PR DESCRIPTION
**Pull request by jumoog**
… when added manually

intro-skipper.org is not available for use in mainland China

## Summary by Sourcery

Improve the plugin configuration by adding a new option to manage the repository URL manually and correct spelling errors in the codebase.

Enhancements:
- Ensure repository URL is only overridden when the OverideManifestUrl option is activated.
- Correct spelling errors in variable names and comments.

### CODE REVIEW

**Comment by jumoog** [comment]
then a new 10.10 docker pull will never get a ` System.IO.IOException: Permission denied` error

**Comment by jumoog** [comment]
@sourcery-ai review

File: IntroSkipper/Plugin.cs [hunk 1]
```diff
@@ -514,14 +513,13 @@ private void InjectSkipButton(string webPath)
...
             {
-                buildVersionString = match.Groups["buildVersion"].Value;
                 webVersionString = match.Groups["webVersion"].Value;
                 _logger.LogInformation("Found jellyfin-web <{WebVersion}>", webVersionString);
                 break;
             }
         }
 
-        if (webVersionString != "unknow")
+        if (webVersionString != "unknown")
```
Comment (AbandonedCart) [inline IntroSkipper/Plugin.cs:522]: I kept not changing this thinking it was intentionally meant to not be unknown

Comment (jumoog) [reply IntroSkipper/Plugin.cs:522]: no just my bad english skills.. the new IDE has a dictionary

**Comment by AbandonedCart** [comment]
The setting makes it easier to avoid typos and keep track of it and all that, but can jellyfin detect a region to automate fixing it?

**Comment by jumoog** [comment]
My first idea was to test if the url is reachable and fall back to github.. but sometimes cloudlfare is down.

**Comment by jumoog** [comment]
@sourcery-ai review

File: IntroSkipper/Plugin.cs [hunk 2]
```diff
@@ -458,16 +458,16 @@ private void MigrateRepoUrl(IServerConfigurationManager serverConfiguration)
...
+            if (pluginRepositories.Exists(repo => repo.Url != null && oldRepos.Contains(repo.Url)))
             {
                 // remove all old plugins
-                pluginRepositories.RemoveAll(repo => repo != null && repo.Url != null && oldRepos.Contains(repo.Url));
+                pluginRepositories.RemoveAll(repo => repo.Url != null && oldRepos.Contains(repo.Url));
 
-                // Add repository only if it does not exit
-                if (!pluginRepositories.Exists(repo => repo.Url == "https://manifest.intro-skipper.org/manifest.json"))
+                // Add repository only if it does not exit and the OverideManifestUrl Option is activated
+                if (!pluginRepositories.Exists(repo => repo.Url == "https://manifest.intro-skipper.org/manifest.json") && Instance!.Configuration.OverrideManifestUrl)
```
Comment (AbandonedCart) [inline IntroSkipper/Plugin.cs:470]: It still may be worthwhile to make this a variable just to separate it

**Review by AbandonedCart — APPROVED** [review]

File: IntroSkipper/Plugin.cs (hunk 2 above)
Comment (jumoog) [reply IntroSkipper/Plugin.cs:470]: But will I ever be able to update the URL again?

Comment (AbandonedCart) [reply IntroSkipper/Plugin.cs:470]: Fair enough and it is only used once, so there's no added risk for typos.

**Pull request merged**